contrib/pkg/deprovision/awstagdeprovision.go

```diff
@@ -65,6 +81,66 @@ func completeAWSUninstaller(o *aws.ClusterUninstaller, logLevel string, args []s
 	return nil
 }
 
+func (o *awsTagDeprovisionOpts) run() error {
+	return wait.ExponentialBackoff(
+		// Start the backoff at 5 minutes, double it each time, to a cap of 24 hours.
+		wait.Backoff{
+			Duration: 5 * time.Minute,
+			Factor:   2,
+			Steps:    1 << 8, // large enough to make cap the effective bound
+			Cap:      24 * time.Hour,
+		},
+		func() (done bool, returnErr error) {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+			defer cancel()
+			uninstaller := &aws.ClusterUninstaller{
+				Filters: o.filters,
+				Logger:  o.logger,
+				Region:  o.region,
+			}
+			blockedResources, err := uninstaller.RunWithContext(ctx)
+			if len(blockedResources) == 0 {
+				return true, err
+			}
+			if o.clusterDeprovision == "" {
+				return
```

Comment:
Do we not also return `err` here even if we will not be posting status to a ClusterDeprovision object?

Reply:
If there are any blocked resources, then we want to keep trying to uninstall in the backoff loop. If we return `err` here, then the uninstall will stop and the pod will fail.

However, maybe it would be good to distinguish between an error because the context expired and an error for other reasons. In the former case, we want to keep trying. In the latter case, we want to abort. I'll look into that.